src/wallet/sqlite.cpp

```diff
@@ -35,6 +36,36 @@ static void ErrorLogCallback(void* arg, int code, const char* msg)
     LogPrintf("SQLite Error. Code: %d. Message: %s\n", code, msg);
 }
 
+static std::optional<int> ReadPragmaInteger(sqlite3* db, const std::string& key, const std::string& description, bilingual_str& error)
+{
+    std::string stmt_text = strprintf("PRAGMA %s", key);
+    sqlite3_stmt* pragma_read_stmt{nullptr};
+    int ret = sqlite3_prepare_v2(db, stmt_text.c_str(), -1, &pragma_read_stmt, nullptr);
+    if (ret != SQLITE_OK) {
+        sqlite3_finalize(pragma_read_stmt);
```

Comment:
Looks like there is still some duplication in the error handling. I wonder if it'd make sense to make RAII wrappers around these things at some point.

Reply:
I agree that RAII wrappers would make sense here to deduplicate the `sqlite3_prepare_v2` / `sqlite3_finalize` pairs. My own C++ skills are probably still too lousy to implement this properly, but I'm happy to help reviewing :)

Reply:
Yes, it was just a random idea, not a blocker for this PR.